### PR TITLE
[ui] Add option to suppress notifications

### PR DESCRIPTION
**Pull request by nephros**
When applying a lot of patches the notification popups get in the way, and clutter the Notification view.

This allows the user to turn off success notifications for applying patches.

### CODE REVIEW

File: src/bin/patchmanager-daemon/patchmanagerobject.cpp [hunk 1]
```diff
@@ -1817,8 +1822,13 @@ void PatchManagerObject::doPatch(const QVariantMap &params, const QDBusMessage &
         }
     }
 
+    // is this used anywhere??
     if (!params.value(QStringLiteral("dont_notify"), false).toBool()) {
-        notify(displayName.toString(), apply ? ok ? NotifyActionSuccessApply : NotifyActionFailedApply : ok ? NotifyActionSuccessUnapply : NotifyActionFailedUnapply);
+        if (getSettings(QStringLiteral("notifyOnSuccess"), true).toBool()) {
```
Comment (b100dian) [inline src/bin/patchmanager-daemon/patchmanagerobject.cpp:1827]: suggestion: `if getSettings(QStringLiteral("notifyOnSuccess"), true).toBool() || !ok` would require one less enum and no return + break:p

Comment (b100dian) [reply src/bin/patchmanager-daemon/patchmanagerobject.cpp:1827]: Wait, the default would be true?

Comment (Olf0) [reply src/bin/patchmanager-daemon/patchmanagerobject.cpp:1827]: WRT the default: Please do not change the extant behaviour!
I.e., all these notifications shall be shown, as long as not deliberately switched off.

Comment (nephros) [reply src/bin/patchmanager-daemon/patchmanagerobject.cpp:1884]: That was the thought behind that default, yes. Keep behaviour, add option to improve (IMO) behaviour.

Comment (b100dian) [reply src/bin/patchmanager-daemon/patchmanagerobject.cpp:1884]: Sorry, true is correct in this case, of course. I got sidetracked.

Comment (nephros) [reply src/bin/patchmanager-daemon/patchmanagerobject.cpp:1884]: > suggestion: `if getSettings(QStringLiteral("notifyOnSuccess"), true).toBool() || !ok` would require one less enum and no return + break:p

Yes - however that logic is (for me) already hard to parse.  
I mean

    if (!params.value(QStringLiteral("dont_notify"), false).toBool())

"if not value dont, (or false) then..." followed by a triple `a ? b : c`...

That return is ugly, yes but I think the enum entry maked it clearer, and it might be reused, e.g. when this is extended to not show other notifications (e.g. one might want to turn off update notifications).

Comment (nephros) [reply src/bin/patchmanager-daemon/patchmanagerobject.cpp]: OK "refactored" a bit according to your suggestion, it's clearer now I think.

Comment (b100dian) [reply src/bin/patchmanager-daemon/patchmanagerobject.cpp]: Ah, yes, no more double?-ternary?-senary? operator, I like it:)

**Review by b100dian — APPROVED** [review]

**Comment by Olf0** [comment]
@nephros, AFAICS this is ready to be merged.

**Comment by Olf0** [comment]
> @nephros, AFAICS this is ready to be merged.

Or is there any reason not to merge this yet?

**Comment by nephros** [comment]
> > @nephros, AFAICS this is ready to be merged.
> 
> Or is there any reason not to merge this yet?

Just inertia, thanks for the reminder. Hitting the green button now! :)
